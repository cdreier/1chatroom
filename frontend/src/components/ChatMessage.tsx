import React from 'react'
import styled from 'styled-components'

interface ContainerProps {
  ownMessage: boolean
}

const Container = styled.div<ContainerProps>`
  border: 1px solid #aeaeae;
  padding: 9px;
  border-bottom-left-radius: ${props => props.ownMessage ? '6px' : '0px'};
  border-bottom-right-radius: ${props => props.ownMessage ? '0px' : '6px'};;
  border-top-left-radius: 6px;
  border-top-right-radius: 6px;
  margin-top: 9px;
  ${props => props.ownMessage ? 'align-self: flex-end;' : ''};
`

const Author = styled.span`
  font-size: 12px;
`

const MessageBody = styled.p`
  margin: 9px 0 0;
`

interface ChatMessageProps {
  author: string
  children: string
  self: string
}

const ChatMessage: React.FC<ChatMessageProps> = ({ children, author, self }) => {
  return (
    <Container ownMessage={self === author}>
      <Author>{author}</Author>
      <MessageBody>{children}</MessageBody>
    </Container>
  )
}

export default ChatMessage
