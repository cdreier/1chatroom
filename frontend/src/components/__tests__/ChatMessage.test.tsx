import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import Theme from '../../theme/default'
import ChatMessage from '../ChatMessage'

describe('ChatMessage', () => {

  test('it renders own messages', () => {
    const sub = jest.fn()
    const component = renderer.create(
      <ThemeProvider theme={Theme}>
        <ChatMessage author={'christian'} self={'christian'} date={new Date(0)} dateFormatter={d => d.toUTCString()}  >
          this is my chat message
        </ChatMessage>
      </ThemeProvider>,
    ).toJSON()
    expect(component).toMatchSnapshot()
  })

  test('it renders other messages', () => {
    const sub = jest.fn()
    const component = renderer.create(
      <ThemeProvider theme={Theme}>
        <ChatMessage author={'stefan'} self={'christian'} date={new Date(500)}  dateFormatter={d => d.toUTCString()}  >
          this is another chat message
        </ChatMessage>
      </ThemeProvider>,
    ).toJSON()
    expect(component).toMatchSnapshot()
  })

})
