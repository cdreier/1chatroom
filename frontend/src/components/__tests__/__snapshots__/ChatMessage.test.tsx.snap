// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessage it renders other messages 1`] = `
.c0 {
  border: 1px solid #eaeaea;
  background-color: #fafafa;
  padding: 15px;
  max-width: 500px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 6px;
  border-top-left-radius: 6px;
  border-top-right-radius: 6px;
  margin-top: 9px;
}

.c1 {
  font-size: 1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: Roboto;
}

.c2 {
  color: #333;
}

.c3 {
  margin-left: 30px;
  color: #333;
}

.c4 {
  margin: 9px 0 0;
  font-family: Playfair Display;
  font-size: 1.2em;
  line-height: 1.65em;
  color: #333;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      stefan
    </span>
    <span
      className="c3"
    >
      1970-1-1
       - 
      1:00:00 AM
    </span>
  </div>
  <p
    className="c4"
  >
    this is another chat message
  </p>
</div>
`;

exports[`ChatMessage it renders own messages 1`] = `
.c0 {
  border: 1px solid #eaeaea;
  background-color: #e8f9f2;
  padding: 15px;
  max-width: 500px;
  border-bottom-left-radius: 6px;
  border-bottom-right-radius: 0px;
  border-top-left-radius: 6px;
  border-top-right-radius: 6px;
  margin-top: 9px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c1 {
  font-size: 1em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: Roboto;
}

.c2 {
  color: #333;
}

.c3 {
  margin-left: 30px;
  color: #333;
}

.c4 {
  margin: 9px 0 0;
  font-family: Playfair Display;
  font-size: 1.2em;
  line-height: 1.65em;
  color: #333;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      christian
    </span>
    <span
      className="c3"
    >
      1970-1-1
       - 
      1:00:00 AM
    </span>
  </div>
  <p
    className="c4"
  >
    this is my chat message
  </p>
</div>
`;
